Pink LSZ Bingo Plugin

EyeKissGuys opened the pull request:
Bingo plugin for our clan for help reduce cheating and automate a bingo event showing bingo tiles info and teams for the bingo event to streamline the process. Sends discord webhooks of screenshots with timestamp to help improve integrity of the bingo board. 

Attaching how the bingo plugin will work / view.

![bingo](https://github.com/user-attachments/assets/e40a1f5c-a310-4aa2-8c8c-5b13335a4ba2)

EyeKissGuys commented:
> Hi! Have you taken a look at the other bingo plugins to see if any of them would work for your use case, maybe with some some configuration / modifications?
> 
> Where possible we'd rather you use something that already exists than build your own, since each new plugin adds extra strain on our review process.

Hello I have, nothing does this in this extent, some send screenshots, but the items file is locked behind other people approval. There is no bingo plugin that extends UI to show progress. This is just a standard plugin, loads a txt file, config a webhook and it handles the rest.

pajlada commented:
What was missing from https://github.com/triplejingle/Bingo-Event-Automated other than the UI? It seems like you can customize it to host your own item list / item sources

EyeKissGuys commented:
> > More based off this one "#8297" which you guys approved no problem
> 
> You are referencing an update PR, there is very clearly the same level of "why do we need another bingo plugin, on the original PR for that plugin as well [#7750 (comment)](https://github.com/runelite/plugin-hub/pull/7750#issuecomment-2773890656)

Yeah i updated the correct pull $ in the comment before you commented. But his bingo plugin was approved for his purposes just trying to figure out the line where its possible to add new features ui teams etc instead of using all these third party services.

It is to more securely handle a bingo event. Unsure how you can make a secure event allowing a front end use to control the items and info being sent. Would love your thoughts.

pajlada commented:
My questions aren't here to reject the plugin, I'm just making you've given thought to the plugin and that you actually need the plugin.

Would you be open to making the item resources URL customizable so other clans can re-use your plugin in the future?

EyeKissGuys commented:
> My questions aren't here to reject the plugin, I'm just making you've given thought to the plugin and that you actually need the plugin.
> 
> Would you be open to making the item resources URL customizable so other clans can re-use your plugin in the future?

I can setup the settings file to be configured and used by every. I more wanted a all in 1 bingo plugin, as some of these use bingo websites to show bingo, then spread sheets to cross references stuff etc, where this is handled by a config file for each team with specific info of how each team is doing on bingo showing real time with a refresh all in 1 place.

EyeKissGuys commented:
> My interrogation has finished - you're now running into build issues. Take a look at the action logs to see what's wrong

Sorry about that was a PL instead of Pl lol. 

I updated the config to accept a settings file for bingo, and added instructions for other users on how to setup the status, board,and teams.

raiyni commented:
A UI sounds like a good contribution to an already existing BINGO plugin.

raiyni commented:
At a certain point, we put a pin in extra bingo plugins. And it looks like https://runelite.net/plugin-hub/show/betterbingo already covers this use case. If you are wanting to show multi-part item progress, that sounds like a good addition.

EyeKissGuys commented:
> At a certain point, we put a pin in extra bingo plugins. And it looks like https://runelite.net/plugin-hub/show/betterbingo already covers this use case. If you are wanting to show multi-part item progress, that sounds like a good addition.

The plugin you sent from a high level pov without looking into it i would see how u can say "already covers this case" this only covers items, not pets, manual validations like experience tiles i.e. gain "10m experience" etc, the teams feature isn't even on runelite yet as they haven't submitted it. this is also an item specific bingo only and does not show real time team progress updates across teams to their board tiles. It just shows basic things from a items based file using. I get we can search "bingo" in plugins but come on let's look into functionality instead too after u install said plugin you reference.

Also to your point "we put a pin in extra bingo plugins" what about the "we put a pin in extra streamer collection log plugins" its the same idea but with another users attempt at the plugin, they are all the same thing "streamer says collection log completed in diff voice", "player plays bingo in diff manner".

raiyni commented:
> The plugin you sent from a high level pov without looking into it i would see how u can say "already covers this case" this only covers items, not pets, manual validations like experience tiles i.e. gain "10m experience" etc, the teams feature isn't even on runelite yet as they haven't submitted it. this is also an item specific bingo only and does not show real time team progress updates across teams to their board tiles. It just shows basic things from a items based file using. I get we can search "bingo" in plugins but come on let's look into functionality instead too after u install said plugin you reference.

Those sound like it would greatly enhance the current plugin and be great additions. To add to that, you said you based this on an already existing plugin that is very actively developed. 

> Also to your point "we put a pin in extra bingo plugins" what about the "we put a pin in extra streamer collection log plugins" its the same idea but with another users attempt at the plugin, they are all the same thing "streamer says collection log completed in diff voice", "player plays bingo in diff manner".

These plugins existed before we started hard encouraging contribution, much like other discord and bingo plugins. But there is a solid difference in face value of knowing you are getting X streamer's voice for X announcements versus `What does this bingo plugin do that this doesn't`. New streamer voice plugins would more than likely be steered towards some form of contribution though.

EyeKissGuys commented:
> > The plugin you sent from a high level pov without looking into it i would see how u can say "already covers this case" this only covers items, not pets, manual validations like experience tiles i.e. gain "10m experience" etc, the teams feature isn't even on runelite yet as they haven't submitted it. this is also an item specific bingo only and does not show real time team progress updates across teams to their board tiles. It just shows basic things from a items based file using. I get we can search "bingo" in plugins but come on let's look into functionality instead too after u install said plugin you reference.
> 
> Those sound like it would greatly enhance the current plugin and be great additions. To add to that, you said you based this on an already existing plugin that is very actively developed.
> 
> > Also to your point "we put a pin in extra bingo plugins" what about the "we put a pin in extra streamer collection log plugins" its the same idea but with another users attempt at the plugin, they are all the same thing "streamer says collection log completed in diff voice", "player plays bingo in diff manner".
> 
> These plugins existed before we started hard encouraging contribution, much like other discord and bingo plugins. But there is a solid difference in face value of knowing you are getting X streamer's voice for X announcements versus `What does this bingo plugin do that this doesn't`. New streamer voice plugins would more than likely be steered towards some form of contribution though.

I would say then you should put plugins you wish people not to create on the README of your plugins github no?

I.e. state there BINGO PLUGINS DO NOT CODE ONLY CONTRIBUTE TO OTHER ONES WE HAVE TOO MANY.

But now since i coded one, it has more features than any of the ones on the plugin hub, that i scrap all code, go try and "maybe" collab with another dev and hope they accept all my pull requests and features maybe they stopped even playing all together and then would need to find a plugin dev that is still current that maybe also has a bingo plugin then refactor it all. You see how much extra time and work that would be?

pajlada commented:
@EyeKissGuys Please have patience as we explore how we want to approach Bingo plugins going forward.

EyeKissGuys commented:
> @EyeKissGuys Please have patience as we explore how we want to approach Bingo plugins going forward.

Yeah all good just trying to pick the brains of contributors as it seems each one has their own opinion on things and i just want to understand their reasoning behind the points they make.

LlemonDuck commented:
@EyeKissGuys

As of writing this comment, we have *SIX* bingo plugins, all of which have ~1k installs or less, in addition to the *FOURTEEN* plugins which can be used to send discord notifications for events, of which bingo could easily be run out of. Each plugin on the hub takes a non-trivial amount of time to review, and incurs future maintenance on us to track you down/fix your plugin if things break.

We would really prefer if you could make an attempt to integrate with an existing plugin before submitting yet another one for our review. The very first section of the plugin-hub README is [Contribute to Existing Plugins https://github.com/runelite/plugin-hub/?tab=readme-ov-file#contribute-to-existing-plugins) for a reason. While I can appreciate wanting to have ownership over the plugin, or being proud of what you wrote and wanting your name attached, submissions like this greatly increase our workload over time. It also leads to increased fragmentation and confusion for our users, which drives support traffic in both discord and github.

If the other plugin author is unresponsive, we can give you ownership of their plugin which would maintain the current installed userbase. If they are unwilling to cooperate, we can revisit this plugin as a new submission.

EyeKissGuys commented:
> @EyeKissGuys
> 
> As of writing this comment, we have _SIX_ bingo plugins, all of which have ~1k installs or less, in addition to the _FOURTEEN_ plugins which can be used to send discord notifications for events, of which bingo could easily be run out of. Each plugin on the hub takes a non-trivial amount of time to review, and incurs future maintenance on us to track you down/fix your plugin if things break.
> 
> We would really prefer if you could make an attempt to integrate with an existing plugin before submitting yet another one for our review. The very first section of the plugin-hub README is [Contribute to Existing Plugins](https://github.com/runelite/plugin-hub/?tab=readme-ov-file#contribute-to-existing-plugins) for a reason. While I can appreciate wanting to have ownership over the plugin, or being proud of what you wrote and wanting your name attached, submissions like this greatly increase our workload over time. It also leads to increased fragmentation and confusion for our users, which drives support traffic in both discord and github.
> 
> If the other plugin author is unresponsive, we can give you ownership of their plugin which would maintain the current installed userbase. If they are unwilling to cooperate, we can revisit this plugin as a new submission.

@LlemonDuck 

I would say imma just close this out then, the time invested and the time it is going to take to get another developer on board is just not worth it.

I just spent 30 minutes reviewing every plugin your Contributors have reviewed and I can bet most of them will also have less than 1k installs, also 1 of them being a bingo plugin update....................................but hey i get it lol.

Thanks and God Speed.
